vignettes/advanced.Rmd

```diff
@@ -26,102 +29,290 @@ knitr::opts_chunk$set(
+- The mean outcome trajectory in the active group was identical to the placebo group up to visit 2. From visit 2 onward, the slope decreased by 50% to 5 points/year.
+- The covariance structure of the baseline and follow-up values in both groups was implied by a random intercept and slope model with a standard deviation of 5 for both the intercept and the slope, and a correlation of 0.25. In addition, an independent residual error with standard deviation 2.5 was added to each assessment.  
+- The probability of study drug discontinuation after each visit was calculated according to a logistic model which also depended on the observed outcome at that visit. Specifically, a visit-wise discontinuation probability of 1.5% and 2.5% in the placebo and active group, respectively, was specified if the observed outcome was 50. For outcomes above 50, the odds of discontinuation further increase by 20% for each 1 point increase. 
+- Study drug discontinuation had no effect on the mean trajectory in the placebo group. In the active group, subjects who discontinued followed the slope of the mean trajectory from the placebo group from that time point onward (CIR). 
+- Study drop-out at the study drug discontinuation visit occurred with a probability of 75% leading to missing outcome data from that time point onward.
+
+This scenario can be implemented as follows:
 
-In order to demonstrate these advanced functions we will first create a simulated dataset
-and perform a mock analysis so that all the require objects are available.
 
 ```{r}
 library(rbmi)
 library(dplyr)
-set.seed(169)
-dat_full <- simulate_test_data(n = 120) %>% as_tibble()
-
-## Introduce missingness
-dat <- dat_full
-missing_index_vis2 <- rbinom(nrow(dat), 1, 0.3) == 1 & dat$visit == "visit_2"
-missing_index_vis3 <- rbinom(nrow(dat), 1, 0.4) == 1 & dat$visit == "visit_3"
-dat[missing_index_vis2, "outcome"] <- NA_real_
-dat[missing_index_vis3, "outcome"] <- NA_real_
-
-# create data_ice setting the imputation method to MAR for
-# each patient with at least one missing value
-dat_ice <- dat %>%
-    arrange(id, visit) %>%
-    filter(is.na(outcome)) %>%
-    group_by(id) %>%
-    slice(1) %>%
-    ungroup() %>%
-    select(id, visit) %>%
-    mutate(strategy = "MAR")
-
-
-# Define the names of key variables in our dataset using `set_vars()`
-# Note that covariates argument can contain interactions
+
+set.seed(135)
+n <- 100
+time <- seq(0, 12, by = 2)
+
+# Mean trajectory control
+muC <- 50 + (10 / 12) * time
+
+# Mean trajectory intervention
+muT <- 50 + (10 / 12) * time - (5 / 12) * pmax(time - 4, 0)
+
+# Create Sigma
+sd_intercept <- 5
+sd_slope <- 5
+cor_slopeInter <- 0.25
+sd_error <- 2.5
+covRE <- matrix(
+    c(
+        sd_intercept^2,
+        cor_slopeInter * sd_intercept * sd_slope,
+        cor_slopeInter * sd_intercept * sd_slope,
+        sd_slope^2
+    ),
+    ncol = 2
+)
+
+Sigma <- cbind(1, time / 12) %*% covRE %*% rbind(1, time / 12) + diag(sd_error^2, nrow = length(time))
```

Comment:
To me this just seems like quite a lot of code which I feel is likely to lose a reader unless they are actively running the code snippits in parallel.  Is it viabe to simplify this at all?  Or maybe instead just hand write the values instead of deriving them so its more explicit to the reader ?

Reply:
I think it would be nice to hand-write some of this, e.g. time, muC, muT, and covRE. I would keep the formula for Sigma.

Reply:
ok, will do it